tests/test_asyncio/test_commands.py

```diff
@@ -2999,6 +3006,37 @@ async def test_module_list(self, r: redis.Redis):
         for x in await r.module_list():
             assert isinstance(x, dict)
 
+    @pytest.mark.onlynoncluster
+    async def test_interrupted_command(self, r: redis.Redis):
+        """
+        Regression test for issue #1128:  An Un-handled BaseException
+        will leave the socket with un-read response to a previous
+        command.
+        """
+        ready = asyncio.Event()
+
+        async def helper():
+            with pytest.raises(asyncio.CancelledError):
+                # blocking pop
+                ready.set()
+                await r.brpop(["nonexist"])
+            # If the following is not done, further Timout operations will fail,
+            # because the timeout won't catch its Cancelled Error if the task
+            # has a pending cancel.  Python documentation probably should reflect this.
+            if sys.version_info.major >= 3 and sys.version_info.minor >= 11:
+                asyncio.current_task().uncancel()
```

Comment:
This looks highly dubious; at least it should never ever be necessary here. Is this a workaround for the stdlib bug fixed in v3.11.3?

Reply:
No, it is necessary in current Python if you decide catch and ignore a Cancel request.
I am the author of pr https://github.com/python/cpython/pull/102815 which fixed the Timeout issue in 3.11.3, and among other things, that PR clarifies this use case.  Please see https://github.com/python/cpython/blob/main/Doc/library/asyncio-task.rst#task-cancellation

Reply:
Oh, I see now. This is necessary since you're catching `CancelledError` above.